bundles/admin/admin-layereditor/view/AdminLayerForm/VisualizationTabPane/RasterStyle.jsx

```diff
@@ -1,68 +1,86 @@
 import React, { Fragment, useState } from 'react';
 import PropTypes from 'prop-types';
-import { Message, Option, TextInput } from 'oskari-ui';
+import { Message, TextInput } from 'oskari-ui';
 import { LocaleConsumer, Controller } from 'oskari-ui/util';
 import { InfoTooltip } from '../InfoTooltip';
-import { LegendImage, GLOBAL_LEGEND } from './LegendImage';
-import { ServiceLegend } from './ServiceLegend';
 import { Link } from './Link';
-import { StyledFormField, Border, DefaultStyle, StyleField, StyleSelect, InlineBlock } from './styled';
+import { StyledFormField, Border, InlineBlock } from './styled';
+import { RasterStyleSelect } from './RasterStyle/RasterStyleSelect';
+import { ServiceLegend } from './RasterStyle/ServiceLegend';
+
+const GLOBAL_LEGEND = 'legendImage';
+
+export const rasterStyleCapabilities = layer => {
+    const { options = {}, capabilities = {}, style } = layer;
+    const { styles = [] } = capabilities;
+    const { legends = {} } = options;
+    const selectedExists = styles.some(s => s.name === style);
+    const warnings = [];
+    if (style && !selectedExists) {
+        warnings.push('defaultStyle');
+    }
+    const additionals = legendsWithoutStyle(styles, legends);
+    const hasGlobal = additionals.find(name => name === GLOBAL_LEGEND);
+    // don't notify if global is only additional legend
+    if (additionals.length > 1 || (additionals.lenth === 1 && !hasGlobal)) {
+        warnings.push('additionalLegend');
+    }
+    if (styles.length > 0 && hasGlobal) {
+        warnings.push('globalWithStyles');
+    }
+    return warnings;
+};
+
+const legendsWithoutStyle = (styles, legends) => {
+    const styleNames = styles.map(s => s.name);
+    return Object.keys(legends)
+        .filter(name => !styleNames.includes(name));
+};
```

Comment:
As `rasterStyleCapabilities()` is only used from handler (outside the component) and `legendsWithoutStyle()` is used by both rasterStyleCapabilities() and the component I would move both of them to some RasterStyle/helper or util.js file. As the functions don't need React at all I think they would benefit from being in a file of their own.